pkg/recordUpdate.go

```diff
@@ -1,11 +1,15 @@
 package pkg
 
+import (
+	"strings"
+)
+
 // UpdateRecord will find all current records and removes them if they are not valid. If no valid records are found it will create a new one.
 func UpdateRecord(domain string, accessToken string, record DNSRecord, records []DNSRecord) error {
 	correctRecordExists := false
 	for _, r := range records {
-		if r.Type == record.Type {
-			if r.Name == record.Name && r.Value == record.Value+"."+domain && !correctRecordExists {
+		if r.Type == record.Type && strings.HasPrefix(r.Name, record.Name) {
+			if r.Value == record.Value {
```

Comment:
`r.Value` and  `record.Value` are the ip addresses. so `record.Value+"."+domain` yields `ip + example.com`, hence records would always be replaced.